server/server.go

```diff
@@ -11,13 +11,13 @@ import (
 )
 
 type Server struct {
-	configuration config.Config
 	httpServer    *http.Server
 }
 
 // NewServer creates a new server configured with the given settings.
 // Start and Stop it with the corresponding functions.
-func NewServer(conf *config.Config) *Server {
+func NewServer() *Server {
```

Comment:
consider keeping the config as a field in the Server struct and passing it into the NewServer method - see my other comments. Could make writing tests harder if this is relying on global variables.

Reply:
I guess with the config.Get() and config.Set() could be enough and we can maintain the NewServer() without config param in the signature.
Do we have an agreement ?

Reply:
+1